frame/babe/src/lib.rs

```diff
@@ -602,6 +617,24 @@ impl<T: Config> Pallet<T> {
 			T::EpochDuration::get(),
 		);
 
+		let current_epoch_index = EpochIndex::<T>::get();
+		if current_epoch_index.saturating_add(1) != epoch_index {
+			// we are skipping epochs therefore we need to update the mapping
+			// of epochs to session
+			if let Some(session_index) = session_index {
+				SkippedEpochs::<T>::mutate(|skipped_epochs| {
+					if skipped_epochs.is_full() {
+						// NOTE: this is O(n) but we currently don't have a bounded `VecDeque`.
+						// this vector is bounded to a small number of elements so performance
+						// shouldn't be an issue.
+						skipped_epochs.remove(0);
```

Comment:
```suggestion
						skipped_epochs.pop();
```

Reply:
pop would remove the most recent entry. We want to remove oldest one

Reply:
weird... I'd assume `remove(index: 0)` removes the first element, same as `pop`